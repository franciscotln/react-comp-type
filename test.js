import React from 'react';
import renderer from 'react-test-renderer';
import Comp from '.';

const Test = ({ text }) => <em>{text}</em>;

test('Comp::contramap - the props are mapped before hitting the component', () => {
  const TestComp = Comp(Test).contramap(props => ({ text: props.value }));

  const component = renderer.create(<TestComp.Render value='contramapped' id='1.618' />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Comp::map - the component should be passed as children to the next mapping component', () => {
  const TestComp = Comp(Test).map(function Parent({ children, text, id }) {
    return (
      <div>
        <p>Original Props</p>
        <p>{`Text: ${text}`}</p>
        <p>{`Id: ${id}`}</p>
        {children}
      </div>
    );
  });

  const component = renderer.create(<TestComp.Render text='mapped' id='2.718' />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Comp::promap - should behave as contramap and map applied together', () => {
  function Parent({ children, value, id }) {
    return (
      <div>
        <p>Original Props</p>
        <p>{`Value: ${value}`}</p>
        <p>{`Id: ${id}`}</p>
        {children}
      </div>
    );
  }

  const TestComp = Comp(Test).promap(props => ({ text: props.value }), Parent);

  const component = renderer.create(<TestComp.Render value='promapped' id='3.141' />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Comp::concat should concatenate two or more components, one after another', () => {
  const TestComp = Comp(Test).concat(Comp(function Another({ text }) {
    return <span>{text}</span>;
  }));

  const component = renderer.create(<TestComp.Render text='concatenating' />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Comp::empty should be a neutral element under concatenation', () => {
  const TestComp = Comp(Test).concat(Comp.empty());

  const component = renderer.create(<TestComp.Render text='concatenating with empty' />);

  expect(component.toJSON()).toMatchSnapshot();
});
