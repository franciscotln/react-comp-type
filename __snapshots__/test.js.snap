// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comp::concat should concatenate two or more components, one after another 1`] = `
Array [
  <em>
    concatenating
  </em>,
  <span>
    concatenating
  </span>,
]
`;

exports[`Comp::contramap - the props are mapped before hitting the component 1`] = `
<em>
  contramapped
</em>
`;

exports[`Comp::empty should be a neutral element under concatenation 1`] = `
<em>
  concatenating with empty
</em>
`;

exports[`Comp::map - the component should be passed as children to the next mapping component 1`] = `
<div>
  <p>
    Original Props
  </p>
  <p>
    Text: mapped
  </p>
  <p>
    Id: 2.718
  </p>
  <em>
    mapped
  </em>
</div>
`;

exports[`Comp::promap - should behave as contramap and map applied together 1`] = `
<div>
  <p>
    Original Props
  </p>
  <p>
    Value: promapped
  </p>
  <p>
    Id: 3.141
  </p>
  <em>
    promapped
  </em>
</div>
`;
